fl4health/privacy_mechanisms/dpPoissonBinomial.py

```diff
@@ -0,0 +1,162 @@
+import math
+from typing import List
+
+import torch
+from scipy.stats import ortho_group
+from torch.distributions import Binomial
+
+
+def ScalerPoissonBinomial(x: float, N: int, radius: float, theta: float) -> int:
+    """
+    Privacy mechanism which encodes scalar data x as a bionomial random variable.
+
+    Args:
+        x (float): Scalar data to be encoded.
+        N (int): Number of Bernoulli trials that yields the binomial variable (We have N = modulus - 1).
+        radius (float): The parameter x must be within the real interval [-radius, radius].
+        theta (float): a parameter within the real interval [0, 1/4].
+
+    Raises:
+        Exception: Invalid parameter N
+        Exception: Invaid radius
+        Exception: Invalid theta
+
+    Returns:
+        int: A random sample taken from the bionomial distribution with success probability
+        p = theta * x / radius + 1/2 and trial number N.
+
+    Reference
+        The Poisson Binomial Mechanism for Unbiased Federated Learning with Secure Aggregation
+        https://proceedings.mlr.press/v162/chen22s/chen22s.pdf
+    """
+
+    if N < 1 or type(N) != int:
+        raise Exception("The parameter N of Binom(N, p) must be a natural number")
+
+    if radius < 0:
+        raise Exception("The radius for the interval must be positive")
+
+    if not (0 <= theta <= 0.25):
+        raise Exception("theta must be within the real interval [0, 1/4]")
+
+    # rescaling
+    p = theta * x / radius + 1 / 2
```

Comment:
Super minor, but you might as well use `theta* x / radius + 0.5` rather than forcing the division here